### PR TITLE
DTS:X Profile 2 Audio support for DASH and fMP4 HLS

### PR DESCRIPTION
**Pull request by Roy-Funderburk**
DTS:X Profile 2 audio sample entries are parsed by MP4Box from input .mp4 files and written to .mp4 output files along with DASH manifest and fMP4 HLS m3u8 files.

### CODE REVIEW

File: include/gpac/constants.h [hunk 1]
```diff
@@ -631,6 +631,9 @@ typedef enum
 
 	GF_CODECID_ALAC = GF_4CC('A','L','A','C'),
 
+	/*! codecid for DTS:X Profile 2 audio */
+	GF_CODECID_DTSX = GF_4CC('d','t','s','x'),
```
Comment (rbouqueau) [inline include/gpac/constants.h:635]: It already existed under the name ```GF_CODECID_DTS_X```.

**Comment by rbouqueau** [comment]
Thank you. I had added some descriptors in c0b48a1936bbbd85cdef00e5813b084d677cb9d5. Which new use-cases does it cover so that we evaluate adding them to our testsuite?

File: include/gpac/constants.h (hunk 1 above)
Comment (Roy-Funderburk) [reply include/gpac/constants.h:635]: I will correct that and upload the fixed version soon.

**Comment by Roy-Funderburk** [comment]
The user can have an input MP4 file containing DTS:X P2 content and use "MP4Box -dash -out output.mpd video.mp4 dtsx.mp4" to create playable DASH output.

Also "MP4Box -dash -ojut output.m3u8 video.mp4 dtsx.mp4" will create fMP4 HLS output.

> Thank you. I had added some descriptors in [c0b48a1](https://github.com/gpac/gpac/commit/c0b48a1936bbbd85cdef00e5813b084d677cb9d5). Which new use-cases does it cover so that we evaluate adding them to our testsuite?

File: src/filters/ff_dec.c [hunk 2]
```diff
@@ -1145,6 +1145,7 @@ static GF_Err ffdec_configure_pid(GF_Filter *filter, GF_FilterPid *pid, Bool is_
 		if (!codec_id) {
 			switch (gpac_codecid) {
 			case GF_CODECID_MVC: codec_id = AV_CODEC_ID_H264; break;
+			case GF_CODECID_DTS_EXPRESS_LBR: codec_id = AV_CODEC_ID_DTS; break;
```
Comment (jeanlf) [inline src/filters/ff_dec.c:1148]: this should be done in ff_common.c in FF2GPAC_CodecIDs array, could you fix that ?

Comment (Roy-Funderburk) [reply src/filters/ff_dec.c]: I've uploaded a new commit, thanks.

**Comment by rbouqueau** [comment]
LGTM. Merging. Thanks!

**Pull request merged**